### ISSUE TITLE
switch to using boot or lein for builds of clojure modules

### ISSUE THREAD

**Issue by loomis** [slipstream/SlipStreamServer#616]
Switch to using `boot` or `lein` for builds of the clojure modules.  Once the switch is made, then clojure dependency management should be moved to a separate file depending on the build tool.  The eastwood and kibit code analyzers should also be implemented as part of the build.

- [ ] switch module builds to `boot` or `lein`
- [ ] switch implementation of central clojure dependency management for build tool 
- [ ] activate eastwood and kibit analyzers as part of build
- [ ] fix issues identified by the code analyzers

**Comment by loomis** [comment]
This is done for the SlipStreamServer/auth module.  The same changes in configuration need to be applied to the other clojure submodules. However this is intrusive and needs to be tested more throughly before being folded into the release.  These changes will be pushed after v2.23 is tagged.

**Comment by konstan** [comment]
- [ ] Client API https://github.com/slipstream/SlipStreamClientAPI/issues/13

**Comment by loomis** [comment]
Holding off until CentOS 7 migration is complete.

**Comment by loomis** [comment]
The builds for SlipStreamServer (auth and ssclj/jar) have been switched to boot.  Similarly the build for SlipStreamServer (ssclj/service-catalog) has also been switched for the enterprise version. 

The build for SlipStreamClientAPI has been updated for boot. This is not currently working because the way the configuration is currently done prevents the compilation of files.  The corresponding branch should be merged only after slipstream/SlipStreamClientAPI#6 has been closed.

All of the branches are called "feature/boot-build".
